llvm/test/tools/sycl-post-link/device-code-split/per-reqd-wg-size-split-2.ll

```diff
@@ -0,0 +1,59 @@
+; The test is intended to check that sycl-post-link correctly groups kernels
+; by unique reqd_work_group_size values used in them
+
+; RUN: sycl-post-link -split=auto -symbols -S %s -o %t.table
+; RUN: FileCheck %s -input-file=%t.table --check-prefix CHECK-TABLE
+;
+; RUN: FileCheck %s -input-file=%t_0.sym --check-prefix CHECK-M0-SYMS \
+; RUN:     --implicit-check-not kernel0 --implicit-check-not kernel1 \
+; RUN:     --implicit-check-not kernel2
+;
+; RUN: FileCheck %s -input-file=%t_1.sym --check-prefix CHECK-M1-SYMS \
+; RUN:     --implicit-check-not kernel1 --implicit-check-not kernel2 \
+; RUN:     --implicit-check-not kernel3
+;
+; RUN: FileCheck %s -input-file=%t_2.sym --check-prefix CHECK-M2-SYMS \
+; RUN:     --implicit-check-not kernel0 --implicit-check-not kernel3
+
+; CHECK-TABLE: Code
+; CHECK-TABLE-NEXT: _0.sym
+; CHECK-TABLE-NEXT: _1.sym
+; CHECK-TABLE-NEXT: _2.sym
+; CHECK-TABLE-EMPTY:
+
+; CHECK-M0-SYMS: kernel3
+
+; CHECK-M1-SYMS: kernel0
+
+; CHECK-M2-SYMS: kernel1
+; CHECK-M2-SYMS: kernel2
+
+target datalayout = "e-i64:64-v16:16-v24:32-v32:32-v48:64-v96:128-v192:256-v256:256-v512:512-v1024:1024"
+target triple = "spir64-unknown-linux"
+
+define dso_local spir_kernel void @kernel0() #0 !sycl_used_aspects !1 {
```

Comment:
This test uses `sycl_used_aspects` metadata and not `reqd_work_group_size` meaning that it checks the wrong thing

Reply:
Exactly... Thanks for finding, fixed.